File: materias ple/programacao orientada ao objeto/trabalhos praticos/tp-1/Codigo/Cpp/Funcionario.cpp
#include "../Hpp/Funcionario.hpp"

Funcionario :: Funcionario(string nome, string cpf, string rg, string endereco, string telefone, string email, string user, string senha, int id) : Pessoa(nome, cpf, rg, endereco, telefone, email), user(user), senha(senha), id(id){}

Funcionario :: ~Funcionario(){};

//Gets

string Funcionario :: getUser(){
    return user;
}

string Funcionario :: getSenha(){
    return senha;
}

int Funcionario :: getId(){
    return id;
}

//Sets

void Funcionario :: setUser(string user){
    this->user = user;
}

void Funcionario :: setSenha(string senha){
    this->senha = senha;
}

void Funcionario :: setId(int id){
    this->id = id;
}

File: materias ple/programacao orientada ao objeto/trabalhos praticos/tp-2/SistemaOficina/src/sistemaoficina/Funcionarios/Funcionario.java
package sistemaoficina.Funcionarios;

import sistemaoficina.Pessoa.Pessoa;

public class Funcionario extends Pessoa{
    private String user;
    private String senha;
    private int id;

    //Construtor
    public Funcionario(String nome, String cpf, String rg, String endereco, String telefone, String email, String user, String senha) {
        super(nome, cpf, rg, endereco, telefone, email);
        setUser(user);
        setSenha(senha);
        this.id = id;
    }

    public Funcionario() {
        this.id = 0;
    }

    //Gets

    public String getUser() {
        return user;
    }

    public String getSenha() {
        return senha;
    }

    public int getId() {
        return id;
    }

    //Sets

    protected void setUser(String user) {
        this.user = user;
    }

    protected void setSenha(String senha) {
        this.senha = senha;
    }
    
    protected void setId(int id) {
        this.id = id;
    }
}